File: common/src/main/java/juuxel/adorn/entity/AdornEntities.java
package juuxel.adorn.entity;

import juuxel.adorn.lib.registry.KeyedRegistrar;
import juuxel.adorn.lib.registry.Registered;
import juuxel.adorn.lib.registry.RegistrarFactory;
import net.minecraft.world.entity.EntityType;
import net.minecraft.world.entity.MobCategory;
import net.minecraft.core.registries.Registries;

public final class AdornEntities {
    public static final KeyedRegistrar<EntityType<?>> ENTITIES = RegistrarFactory.get().create(Registries.ENTITY_TYPE);

    public static final Registered<EntityType<SeatEntity>> SEAT = ENTITIES.register("seat",
        key -> EntityType.Builder.of(SeatEntity::new, MobCategory.MISC)
            .sized(0f, 0f)
            .build(key));

    public static final Registered.WithKey<EntityType<?>, EntityType<ConeEntity>> CONE = ENTITIES.register("cone",
        key -> EntityType.Builder.of(ConeEntity::new, MobCategory.MISC)
            .noLootTable()
            .sized(0.75f, 0.875f)
            .clientTrackingRange(10)
            .build(key));

    public static void init() {
    }
}

File: common/src/main/kotlin/juuxel/adorn/entity/AdornEntities.kt
package juuxel.adorn.entity

import juuxel.adorn.lib.registry.Registrar
import juuxel.adorn.lib.registry.RegistrarFactory
import juuxel.adorn.platform.PlatformBridges
import net.minecraft.entity.EntityType
import net.minecraft.registry.RegistryKeys

object AdornEntities {
    @JvmField
    val ENTITIES: Registrar<EntityType<*>> = RegistrarFactory.get().create(RegistryKeys.ENTITY_TYPE)

    val SEAT: EntityType<SeatEntity> by ENTITIES.register("seat", PlatformBridges.entities::createSeatType)

    fun init() {}
}
